lib/karafka/base_controller.rb

```diff
@@ -100,18 +36,32 @@ def params_batch=(messages)
     # Executes the default controller flow, runs callbacks and if not halted
     # will call process method of a proper backend
     def call
-      run_callbacks :after_received do
-        process
-      end
+      process
     end
 
     private
 
-    # Method that will perform business logic on data received from Kafka
+    # We make it private as it should be accesible only from the inside of a controller
+    attr_reader :params_batch
+
+    # @return [Karafka::Connection::Consumer] messages consumer that can be used to
+    #    commit manually offset or pause / stop consumer based on the business logic
+    def consumer
+      Persistence::Consumer.read
+    end
+
+    # Method that will perform business logic and on data received from Kafka (it will consume
+    #   the data)
     # @note This method needs bo be implemented in a subclass. We stub it here as a failover if
     #   someone forgets about it or makes on with typo
-    def perform
+    def consume
```

Comment:
Great idea. This is way clearer.

Reply:
Yeah and added a deprecation warning + sidekiq-karafka supports both